src/lem/circuit.rs

```diff
@@ -48,30 +48,21 @@ use crate::{
 
 use super::{
     interpreter::{Frame, PreimageData},
-    pointers::{Ptr, ZPtr},
+    pointers::Ptr,
     slot::*,
     store::Store,
     var_map::VarMap,
-    Block, Ctrl, Func, Op, Tag, Var,
+    Block, Ctrl, Func, Op, Tag,
 };
 
 /// Manages global allocations for constants in a constraint system
 #[derive(Default)]
 pub struct GlobalAllocator<F: LurkField>(HashMap<FWrap<F>, AllocatedNum<F>>);
 
-#[inline]
-fn allocate_num<F: LurkField, CS: ConstraintSystem<F>>(
-    cs: &mut CS,
-    namespace: &str,
-    value: F,
-) -> Result<AllocatedNum<F>, SynthesisError> {
-    AllocatedNum::alloc(cs.namespace(|| namespace), || Ok(value))
-}
-
 impl<F: LurkField> GlobalAllocator<F> {
     /// Checks if the allocation for a numeric variable has already been cached.
     /// If so, don't do anything. Otherwise, allocate and cache it.
-    fn alloc_const<CS: ConstraintSystem<F>>(&mut self, cs: &mut CS, f: F) {
+    fn insert_allocate_const<CS: ConstraintSystem<F>>(&mut self, cs: &mut CS, f: F) {
```

Comment:
I don't like this long name very much

Reply:
I changed the name because it is confusing considering we already have an `allocate_const`. Maybe `new_const` is a better name?